src/actors/cficaches.rs

```diff
@@ -168,14 +182,38 @@ impl CacheItemRequest for FetchCfiCacheInternal {
         data: ByteView<'static>,
         path: CachePath,
     ) -> Self::Item {
+        let unwind = match status {
+            CacheStatus::Positive => ObjectUseInfo::Ok,
+            CacheStatus::Negative => {
+                if self.meta_handle.status() == CacheStatus::Positive {
+                    ObjectUseInfo::Error {
+                        details: String::from("Object file no longer available"),
+                    }
+                } else {
+                    // No need to pretend that we were going to use this cficache if the
+                    // original object file was already not there, that status is already
+                    // reported.
+                    ObjectUseInfo::None
+                }
+            }
+            CacheStatus::Malformed => ObjectUseInfo::Malformed,
+        };
```

Comment:
_nit_: This block could be moved into a function. This would be great since this will make unit testing it easier and we can add a descriptive doc comment that describes the edge cases it handles.